### PR TITLE
Add ability to copy large files

### PR DESCRIPTION
**Pull request by protsack-stephan**
Fix for the `Copy` method that will allow to copy large files more than `5GB`.

### CODE REVIEW

**Comment by memije** [comment]
LGTM

File: lib/s3/storage.go [hunk 1]
```diff
@@ -185,10 +188,75 @@ func (s *Storage) Copy(src string, dst string, options ...map[string]interface{}
...
+	if *hr.ContentLength <= maxUploadSizeBytes {
+		_, err = s.s3.CopyObject(&s3.CopyObjectInput{
+			Bucket:     aws.String(bucket),
+			CopySource: aws.String(fmt.Sprintf("%s/%s", s.bucket, src)),
+			Key:        aws.String(dst),
+		})
+
+		if err != nil {
+			return err
+		}
```
Comment (prabhat393) [inline lib/s3/storage.go]: I think we should return nil here for the case of no err.

File: lib/s3/storage.go [hunk 2]
```diff
@@ -185,10 +188,75 @@ func (s *Storage) Copy(src string, dst string, options ...map[string]interface{}
...
+	})
+
+	if err != nil {
+		return err
+	}
+
+	cmu := &s3.CompletedMultipartUpload{}
+	maxPart := int(math.Ceil(float64(*hr.ContentLength) / float64(maxUploadSizeBytes)))
+
+	for prt := 0; prt < int(maxPart); prt++ {
```
Comment (prabhat393) [inline lib/s3/storage.go]: Do we need to cast maxPart here? It's already casted to int above.

File: lib/s3/storage.go [hunk 3]
```diff
@@ -207,10 +275,75 @@ func (s *Storage) CopyWithContext(ctx aws.Context, src string, dst string, optio
...
+	if *hr.ContentLength <= maxUploadSizeBytes {
+		_, err = s.s3.CopyObjectWithContext(ctx, &s3.CopyObjectInput{
+			Bucket:     aws.String(bucket),
+			CopySource: aws.String(fmt.Sprintf("%s/%s", s.bucket, src)),
+			Key:        aws.String(dst),
+		})
+
+		if err != nil {
+			return err
+		}
```
Comment (prabhat393) [inline lib/s3/storage.go]: Same comment

File: lib/s3/storage.go [hunk 4]
```diff
@@ -207,10 +275,75 @@ func (s *Storage) CopyWithContext(ctx aws.Context, src string, dst string, optio
...
+	})
+
+	if err != nil {
+		return err
+	}
+
+	cmu := &s3.CompletedMultipartUpload{}
+	maxPart := int(math.Ceil(float64(*hr.ContentLength) / float64(maxUploadSizeBytes)))
+
+	for prt := 0; prt < int(maxPart); prt++ {
```
Comment (prabhat393) [inline lib/s3/storage.go]: Same comment.

**Comment by protsack-stephan** [comment]
@prabhat393 Fixed the comments.

**Review by prabhat393 — APPROVED** [review]

**Comment by prabhat393** [comment]
Looks good.

**Pull request merged**